server/src/main/java/io/druid/segment/realtime/appenderator/BatchAppenderatorDriver.java

```diff
@@ -132,14 +134,40 @@ private SegmentsAndMetadata pushAndClear(
         .map(SegmentWithState::getSegmentIdentifier)
         .collect(Collectors.toList());
 
-    final ListenableFuture<SegmentsAndMetadata> future = Futures.transform(
-        pushInBackground(null, segmentIdentifierList),
-        this::dropInBackground
-    );
+    final CompletableFuture<SegmentsAndMetadata> chainEndFuture = new CompletableFuture<>();
```

Comment:
And here

Reply:
If `SettableFuture` is used instead of `CompletableFuture`, this code becomes isomorphic to the code in `AppenderatorDriverRealtimeIndexTask` and a common method `transform()` (replacement of `Futures.transform()`) could be extracted.